components/dashboard/src/settings/PersonalAccessTokens.tsx

```diff
@@ -417,12 +418,15 @@ interface TokenEntryProps {
 }
 
 function TokenEntry(props: TokenEntryProps) {
-    const getDate = () => {
+    const [dateInfo, setDateInfo] = useState<{ expired: boolean; content: string }>();
+
+    useEffect(() => {
```

Comment:
No need to use `useEffect`, just calculate `const expired = dayjs(props.token.expirationTime!.toDate()).isBefore(dayjs());` as `TokenEntry` is stateless

Reply:
Will `props.token` change update expired too?